plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1753,4 +1756,381 @@ private Set<String> getAllFilesFromCdfDirectory(String tableName)
+                "test_partition_filter_removed_",
+                "(x VARCHAR, part VARCHAR) WITH (PARTITIONED_BY = ARRAY['part'])",
+                ImmutableList.of("('a', 'part_a')", "('b', 'part_b')"))) {
+            assertQueryFails(session, "SELECT x FROM " + table.getName() + " WHERE part IS NOT NULL OR TRUE", "Filter required on .*" + table.getName() + " for at least one partition column:.*");
+        }
+    }
+
+    @Test
+    public void testPartitionFilterIncluded()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_filter_included",
+                "(x VARCHAR, part INT) WITH (PARTITIONED_BY = ARRAY['part'])",
+                ImmutableList.of("('a', 1)", "('a', 2)", "('a', 3)", "('a', 4)", "('b', 1)", "('b', 2)", "('b', 3)", "('b', 4)"))) {
+            assertQuery(session, "SELECT * FROM " + table.getName() + " WHERE part = 1", "VALUES ('a', 1), ('b', 1)");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part < 2", "VALUES 2");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE Part < 2", "VALUES 2");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE PART < 2", "VALUES 2");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE parT < 2", "VALUES 2");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part % 2 = 0", "VALUES 4");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part - 2 = 0", "VALUES 2");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part * 4 = 4", "VALUES 2");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part % 2 > 0", "VALUES 4");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part % 2 = 1 and part IS NOT NULL", "VALUES 4");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part IS NULL", "VALUES 0");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part = 1 OR x = 'a' ", "VALUES 5");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part = 1 AND  x = 'a' ", "VALUES 1");
+            assertQuery(session, "SELECT count(*) FROM " + table.getName() + " WHERE part IS NOT NULL", "VALUES 8");
+            assertQuery(session, "SELECT x, count(*) AS COUNT FROM " + table.getName() + " WHERE part > 2 GROUP BY x ", "VALUES ('a', 2), ('b', 2)");
+            assertQueryFails(session, "SELECT count(*) FROM " + table.getName() + " WHERE x= 'a'", "Filter required on .*" + table.getName() + " for at least one partition column:.*");
+        }
+    }
+
+    @Test
+    public void testPartitionFilterIncludeCaseSensitivity()
```

Comment:
I'm not sure if this test would help us there - Trino would consider the column name to lower case irrespective of the case - I think we need to compare with the table created in delta with a different column case.

Reply:
Ok, already covered by test in TestDeltaLakeRequireQueryPartitionsFilter.testRequiresQueryPartitionFilterWithUppercaseColumnName